x/gov/types.go

```diff
@@ -0,0 +1,82 @@
+package gov
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+)
+
+var (
+	proposalTypes = []string{"Text", "ParameterChange", "SoftwareUpgrade"}
+)
+
+// Vote
+type Vote struct {
+	Voter      sdk.Address `json:"voter"`       //  address of the voter
+	ProposalID int64       `json:"proposal_id"` //  proposalID of the proposal
+	Option     string      `json:"option"`      //  option from OptionSet chosen by the voter
+}
+
+//-----------------------------------------------------------
+
+// Proposal
+type Proposal struct {
+	ProposalID   int64  `json:"proposal_id"`   //  ID of the proposal
+	Title        string `json:"title"`         //  Title of the proposal
+	Description  string `json:"description"`   //  Description of the proposal
+	ProposalType string `json:"proposal_type"` //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+
+	Status string `json:"string"` //  Status of the Proposal {Pending, Active, Passed, Rejected}
```

Comment:
Can we use an enum (uint8)?

Reply:
For integration into apps a string representation could be helpful.

Reply:
Agree with Fabo.  Also, how I do ProposalTypes might change (have multiple dimensions), so I think keeping it as a string is better for now, and can optimize this later.

Reply:
String representations can absolutely be returned by REST handlers, we can add a function to turn an enum into a string - but representing as enums is better IMO:

- More efficient storage & equality comparisions
- Prevents using an invalid string value by accident